types/micromatch/index.d.ts

```diff
@@ -266,6 +266,60 @@ declare namespace micromatch {
          */
         windows?: boolean;
     }
+
+    interface ScanOptions extends Options {
+        /**
+         * When `true`, the returned object will include an array of `tokens` (objects), representing each path "segment" in the scanned glob pattern.
+         *
+         * @default false
+         */
+        tokens?: boolean;
+        /**
+         * When `true`, the returned object will include an array of strings representing each path "segment" in the scanned glob pattern.
+         * This is automatically enabled when `options.tokens` is `true`.
+         *
+         * @default false
+         */
+        parts?: boolean;
+    }
+
+    interface ScanInfo {
+        prefix: string;
+        input: string;
+        start: number;
+        base: string;
+        glob: string;
+        isBrace: boolean;
+        isBracket: boolean;
+        isGlob: boolean;
+        isExtglob: boolean;
+        isGlobstar: boolean;
+        negated: boolean;
+    }
+
+    interface ScanInfoToken {
```

Comment:
JFYI

https://github.com/micromatch/picomatch/blob/66e1b200f3ea1784bdeae0770c8e5cd95f55e334/lib/scan.js#L73

Other fields can be found by search: `token.`